src/Common.Lib/refcount.cpp

```diff
@@ -70,27 +72,26 @@ namespace MicrosoftInstrumentationEngine
 
             va_list marker;
             va_start(marker, pszFormat);
-            _vsnwprintf_s(szTrace + offset, ARRAYSIZE(szTrace) - offset, _TRUNCATE, pszFormat, marker);
-            wcscat_s(szTrace, ARRAYSIZE(szTrace), _T("\r\n"));
-            OutputDebugStringW(szTrace);
+            _vsnwprintf_s(szTrace.data() + offset, szTrace.size() - offset, _TRUNCATE, pszFormat, marker);
+            wcscat_s(szTrace.data(), szTrace.size(), _T("\r\n"));
+            OutputDebugStringW(szTrace.data());
         }
     }
-#endif
 
     // ----------------------------------------------------------------------------
     // CRefRBMap
-#ifdef DEBUG
-    class CRefRBMap : public ATL::CRBMap<CRefCount*, ULONG> {};
+    class CRefRBMap : public std::map<CRefCount*, ULONG> {};
```

Comment:
I'm assuming our implementation here diverges from the original because we don't want to bring in atl/pal into the common library?

Reply:
Yes, exactly. If we use PAL calls in this lib, it means that it can't be used in multiple dlls on musl systems because the dynamic linker will get name clashes. I would like to build the Nagler instrumentation method without a dependency on the PAL.

---
In reply to: [573075612](https://github.com/microsoft/CLRInstrumentationEngine/pull/362#discussion_r573075612) [](ancestors = 573075612)